src/main/java/io/orkes/conductor/client/spring/OrkesSpringWorkerConfiguration.java

```diff
@@ -28,19 +28,26 @@ public OrkesSpringWorkerConfiguration(Environment environment) {
 
     @Override
     public int getPollingInterval(String taskName) {
-        String key = "conductor.worker." + taskName + ".pollingInterval";
-        return environment.getProperty(key, Integer.class, 0);
+        return getProperty(taskName, "pollingInterval", Integer.class, 0);
     }
 
     @Override
     public int getThreadCount(String taskName) {
-        String key = "conductor.worker." + taskName + ".threadCount";
-        return environment.getProperty(key, Integer.class, 0);
+        return getProperty(taskName, "threadCount", Integer.class, 0);
     }
 
     @Override
     public String getDomain(String taskName) {
-        String key = "conductor.worker." + taskName + ".domain";
-        return environment.getProperty(key, String.class, null);
+        return getProperty(taskName, "domain", String.class, null);
+    }
+
+    private <T>T getProperty(String taskName, String property, Class<T> type, T defaultValue) {
+        String key = "conductor.worker." + taskName + "." + property;
+        T value = environment.getProperty(key, type, defaultValue);
+        if(value == null || value == defaultValue) {
```

Comment:
Can the `value` ever be null, now that you have a defaultValue?

Reply:
value can be null if not set.